src/berry_mill/kiwrap.py

```diff
@@ -107,9 +120,17 @@ def _get_repokeys(self, reponame: str, repodata: Dict[str, str]) -> str:
         else:
             s_url = urljoin(f"{url.scheme}://{url.netloc}/{url.path}/Release.key", "")
             response = requests.get(s_url, allow_redirects=True)
-            with open(g_path, 'xb') as f_rel:
-                f_rel.write(response.content)
-            response.close()
+            # check reponse OK
+            if response.status_code == HTTPStatus.OK:
+                with open(g_path, 'xb') as f_rel:
+                    f_rel.write(response.content)
+                response.close()
+            else:
+                raise SystemExit(f"Wrong url defined for repo {reponame}")
+
+            if not self._verify_gpg_key(g_path):
+                log.warning(f"GPG key check failed for repo {reponame}")
+                os.remove(g_path)
```

Comment:
please set the g_path also to "" afterwards. Otherwise you will return a g_path pointing to a file that no longer exists and berrymill will fail at another (wrong) point in the code.